lib/ansible/plugins/lookup/filetree.py

```diff
@@ -100,6 +100,22 @@ def file_props(root, path):
 
     return ret
 
+def walk(path, topdown=None):
+    topdown = topdown or False
+
+    ret = []
+    for root, dirs, files in os.walk(path, topdown=True):
+        for entry in dirs + files:
+            relpath = os.path.relpath(os.path.join(root, entry), path)
+
+            # Skip if relpath was already processed (from another root)
+            if relpath in [entry['path'] for entry in ret]:
```

Comment:
I think my version is buggy as well, since the ```entry['path'] for entry in ret``` in walk() will only check the local 'ret', which will be per term, where the original check against ret would see all results.

I could pass the run() level 'ret' to walk() and let walk modify it accumulating results. And something
similar for a 'visited_paths' collection. 

I sketched out a generator based version that could work, but is currently unfinished.